src/openssl/ssl/context.cr

```diff
@@ -108,6 +108,10 @@ abstract class OpenSSL::SSL::Context
   end
 
   class Server < Context
+    # Keep a reference so the GC doesn't collect it after sending it to C land
+    @sni_callback : Proc(String, OpenSSL::SSL::Context::Server?)?
+    @sni_callback_box = Pointer(Void).null
```

Comment:
I don't think it's necessary to keep a reference to both, the box and its value.
The box should be sufficient: In case of a pointer type, it's identical to its value, and in case of a struct, the box allocates it on the heap anyway.
```suggestion
    # Keep a reference so the GC doesn't collect it after sending it to C land
    @sni_callback_box = Pointer(Void).null
```